invokeai/backend/model_manager/load/model_cache/model_cache_default.py

```diff
@@ -267,6 +268,13 @@ def move_model_to_device(self, cache_entry: CacheRecord[AnyModel], target_device
         start_model_to_time = time.time()
         snapshot_before = self._capture_memory_snapshot()
         try:
+            if target_device == self.storage_device:
+                cache_entry.model.load_state_dict(cache_entry.state_dict, assign=True)
+            else:
+                new_dict: Dict[str, torch.Tensor] = {}
+                for k, v in cache_entry.state_dict.items():
+                    new_dict[k] = v.to(torch.device(target_device), copy=True)
+                cache_entry.model.load_state_dict(new_dict, assign=True)
```

Comment:
We should add a comment to this block to explain why we are doing this. Otherwise, this will seem like a very roundabout way of moving models to someone without the full context.

Reply:
> We should add a comment to this block to explain why we are doing this. Otherwise, this will seem like a very roundabout way of moving models to someone without the full context.

Done.